rubix/core/data.py

```diff
@@ -262,18 +310,29 @@ def prepare_input(config: Union[dict, str]) -> RubixData:
     # Load the data from the file
     data, units = load_galaxy_data(file_path)
 
+    # Create the RubixData object
     rubixdata = RubixData(Galaxy(), StarsData(), GasData())
 
+    # Set the galaxy attributes
     rubixdata.galaxy.redshift = data["redshift"]
+    rubixdata.galaxy.redshift_unit = units["galaxy"]["redshift"]
```

Comment:
Not sure if I like this approahc with storing the units inside an attribute under `attribute_unit`. An alternative would be to save units as a dictionary like 

```python
rubixdata.galaxy.units = {
    "redshift": units["galaxy"]["redshift"],
    "center": units["galaxy"]["center"],
    "halfmassrad_stars": units["galaxy"]["halfmassrad_stars"],
}
``` 
in the future this can be replaced with other package that implement JAX unit support

Reply:
but this is just preference

Reply:
let me think about this.
I see some advantages for the units dictionary for future extensibility and ease of maintainability.